Lib/test/test_types.py

```diff
@@ -666,6 +666,13 @@ def test_or_type_operator_with_TypeVar(self):
         assert TV | str == typing.Union[TV, str]
         assert str | TV == typing.Union[str, TV]
 
+    def test_union_parameter_chaining(self):
+        T = typing.TypeVar("T")
+
+        self.assertEqual((float | list[T])[int], float | list[int])
+        self.assertEqual(list[int | list[T]].__parameters__, (T,))
+        self.assertEqual(list[int | list[T]][str], list[int | list[str]])
```

Comment:
Add a check showing that having S and T in different parts gets unified? E.g. `list[T] | list[S] --> __parameters__ == (T, S)` and if we substitute something crazy like `[int, T]` it becomes `list[int] | list[T]`.

Reply:
Tests have been added, thanks for suggestion